### PR TITLE
Update documentation on infrastructure & cluster provisioning

### PR DESCRIPTION
**Pull request by nikimanoledaki**
#### What type of PR is this?

Adds documentation so that we can stop using the design doc.

#### What this PR does / why we need it:

- Moves `docs/` to `docs/infrastructure`
- Adds Roadmap to README
- Add info about infra / cluster / component provisioning, monitoring, access

#### Which issue(s) this PR fixes:

https://github.com/cncf-tags/green-reviews-tooling/issues/47

### CODE REVIEW

File: docs/cncf-projects/README.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+## CNCF Project Maintainer Responsibilities
+
+There are different roles and responsibilities involved in the review of the CNCF Projects. What do CNCF Project Maintainers provide? What does the Green Reviews WG provide? These questions are being answered through the current active collaboration with the maintainers of Falco.
+
+There are certain key differences around the deployment and implementation that vary for each project. CNCF Project Maintainers can help with the following:
+- Provide a way to install the CNCF Project in the community cluster and c
+- Share any requirements for deploying the CNCF Project or running the benchmarks
+- Contribute test scenarios for each of the specific CNCF Project, which should roughly be equivalent to the [functional unit](https://sci-guide.greensoftware.foundation/R) of the tool
```
Comment (rossf7) [inline docs/cncf-projects/README.md:12]:
```suggestion
- Contribute test scenarios for a specific CNCF Project, which should roughly be equivalent to the [functional unit](https://sci-guide.greensoftware.foundation/R) of the tool's SCI calculation
```

File: docs/cncf-projects/falco/README.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+# Falco
+
```
Comment (rossf7) [inline docs/cncf-projects/falco/README.md:2]:
```suggestion
TBC See https://github.com/falcosecurity/cncf-green-review-testing
```

I think we can add this as a follow up but we could link to the repo as a starting point.

File: docs/infrastructure/README.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    - [Cluster Access](#cluster-access)
+  - [Monitoring](#monitoring)
+
+## Provisioning Infrastructure & Cluster
+
+The infrastructure used for this project leverages infrastructure credits that Equinix kindly donated to the CNCF to support open-source projects.
+
+The Green Reviews WG uses Infrastructure as Code (IaC) tools as much as possible. For example, tools used in this project include [OpenTofu](https://opentofu.org/) and [k3s](https://k3s.io/) to provision the infrastructure and cluster. Flux is used to provision the cluster components (see [Cluster Management](#cluster-management) below).
+
+OpenTofu reconciles the files in [infrastructure/](../../infrastructure/):
```
Comment (rossf7) [inline docs/infrastructure/README.md:16]:
```suggestion
OpenTofu reconciles the files in [infrastructure/](../../infrastructure/) via this GitHub [action](https://github.com/cncf-tags/green-reviews-tooling/blob/main/.github/workflows/tofu.yaml):
```

File: docs/infrastructure/README.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+```bash
+./infrastructure
+└── equinix-metal
+    ├── main.tf
+    └── variables.tf
+```
+
+OpenTofu provisions the Equinix Metal infrastructure using the [Equinix Terraform provider](https://github.com/equinix/terraform-provider-equinix). `user_data` provisions Kubernetes with k3s. It bootstrap the control plane, worker nodes, Cilium CNI, Flux (see below), etc.
```
Comment (rossf7) [inline docs/infrastructure/README.md:24]:
```suggestion
OpenTofu provisions the Equinix Metal infrastructure using the [Equinix Terraform provider](https://github.com/equinix/terraform-provider-equinix). `user_data` provisions Kubernetes with k3s. It bootstraps the control plane, worker nodes, Cilium CNI, Flux (see below), etc.
```

**Comment by nikimanoledaki** [comment]
@rossf7 thank you for the review! 😄  Implemented your feedback 👍

**Review by rossf7 — APPROVED** [review]
LGTM

**Pull request merged**